### ISSUE TITLE
worker update failed?

### ISSUE THREAD

**Issue by vondele** [glinscott/fishtest#1166]
noobpwnftw:

```
Worker started in /home/fishtest/fishtest-master/worker ... (PID=24939)
Creating lock file /home/fishtest/fishtest-master/worker/worker.lock
Memory: 3847155712
Config file /home/fishtest/fishtest-master/worker/fishtest.cfg written
Found g++ version 9.3.1
UUID: a1205cf9-f700-4140-a159-65bce98f35f9
Worker version 131 connecting to http://tests.stockfishchess.org:80
Start heartbeat
API call rate limits: {'limit': 5000, 'used': 0, 'remaining': 5000, 'reset': 1637960730}
Verify worker version...
Incorrect username/password
Waiting for the heartbeat thread to finish...
Heartbeat stopped
Deleting lock file /home/fishtest/fishtest-master/worker/worker.lock
```

my worker version is 131 and it won't upgrade
it deadloops the above
very few brave workers managed to upgrade to 132
and those works fine

**Comment by vdbergh** [comment]
I can reproduce it.

**Comment by vdbergh** [comment]
I have made a PR to fix this. The server is now much more strict about accepting client requests. But of course the worker needs to be adapted to this. And for that it has to be updated first....

**Comment by ppigazzini** [comment]
Fixed by #1167, PROD updated.
